mlflow/utils/model_utils.py

```diff
@@ -188,10 +250,21 @@ def _add_code_from_conf_to_system_path(local_path, conf, code_key=FLAVOR_CONFIG_
             By default this is FLAVOR_CONFIG_CODE.
     """
     assert isinstance(conf, dict), "`conf` argument must be a dict."
+
+    modules_to_reload = set()
     if code_key in conf and conf[code_key]:
         code_path = os.path.join(local_path, conf[code_key])
         _add_code_to_system_path(code_path)
 
+        for loaded_module in sys.modules:
+            if os.path.exists(os.path.join(code_path, loaded_module.replace('.', os.sep))):
+                # this is a module loaded from code path
+                modules_to_reload.add(loaded_module)
+
+    # invalidate these modules cache, to ensure they are reloaded when loading model.
+    for module in modules_to_reload:
+        sys.modules.pop(module)
```

Comment:
```suggestion
        sys.modules.pop(module, None)
```

just to make the code look safe.